internal/engine/compiler/impl_amd64.go

```diff
@@ -2276,13 +2250,13 @@ func (c *amd64Compiler) emitUnsignedI32TruncFromFloat(isFloat32Bit, nonTrapping
 	okJmpForAboveOrEqualMaxInt32PlusOne := c.assembler.CompileJump(amd64.JMP)
 
 	c.assembler.SetJumpTargetOnNext(jmpIfPlusInf)
-	if !nonTrapping {
-		c.compileExitFromNativeCode(nativeCallStatusIntegerOverflow)
-	} else {
+	if nonTrapping {
 		err = c.assembler.CompileStaticConstToRegister(amd64.MOVL, c.maximum32BitUnsignedInt, result)
 		if err != nil {
 			return err
 		}
+	} else {
+		c.compileExitFromNativeCode(nativeCallStatusIntegerOverflow)
```

Comment:
I skipped this one since I couldn't make sense of the above logic, but can try a bit harder if you want me to.